dashboard/app/models/user.rb

```diff
@@ -2052,6 +2057,14 @@ def show_census_teacher_banner?
     teacher? && users_school && (next_census_display.nil? || Date.today >= next_census_display.to_date)
   end
 
+  # Returns the name of the donor for the donor teacher banner, or nil if none.
+  def donor_teacher_banner_name
+    school_id = school_info&.school&.id
```

Comment:
We shouldn't use `school_info` anymore, now that we're tracking school history for teachers.  Probably more correct to say
```suggestion
    school_id = last_complete_school_info&.school&.id
```
@bethanyaconnor @nkiruka to check that I'm remembering this correctly.

Reply:
I don't think we ended up deprecating `school_info` because it would've caused problems for the data team but @nkiruka can confirm.

Reply:
Yes, we should use the later (last_complete_school_info) and we haven't deprecated school_info yet.